### PR TITLE
pygmt.datasets.load_*: Add autocompletion support for the 'resolution' parameter

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds type hints to the `resolution` parameter of the `load_**` functions, so that editors can autocomplete the `resolution` parameter.

**Preview**:https://pygmt-dev--3260.org.readthedocs.build/en/3260/api/generated/pygmt.datasets.load_earth_relief.html

image

- Pros: autocompletion for the `resolution` parameter.
- Cons: The function definition is much longer.

@GenericMappingTools/pygmt-maintainers What do you think?

### CODE REVIEW

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers for comments.

File: pygmt/datasets/earth_relief.py [hunk 1]
```diff
@@ -16,7 +16,23 @@
...
+        "05m",
+        "04m",
+        "03m",
+        "02m",
+        "01m",
+        "30s",
+        "15s",
+        "03s",
+        "01s",
+    ] = "01d",
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:35]: Or we can write the long list in a single line and let `ruff` ignore it so that ruff won't reformat it.

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/earth_magnetic_anomaly.py [hunk 2]
```diff
@@ -64,11 +66,9 @@ def load_earth_magnetic_anomaly(
...
     Parameters
     ----------
-    resolution : str
-        The grid resolution. The suffix ``d`` and ``m`` stand for
-        arc-degrees and arc-minutes. It can be ``"01d"``, ``"30m"``,
-        ``"20m"``, ``"15m"``, ``"10m"``, ``"06m"``, ``"05m"``, ``"04m"``,
-        ``"03m"``, or ``"02m"``. The ``"02m"`` resolution is not available for
+    resolution
+        The grid resolution. The suffix ``d`` and ``m`` stand for arc-degrees and
+        arc-minutes. The ``"02m"`` resolution is not available for
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_magnetic_anomaly.py:71]:
```suggestion
        The grid resolution. The suffix ``d`` and ``m`` stand for arc-degrees and
        arc-minutes. The resolution ``"02m"`` is not available for
```

File: pygmt/datasets/earth_relief.py (hunk 1 above)
Comment (yvonnefroehlich) [reply pygmt/datasets/earth_relief.py:35]: Hm. Looks a bit weird. I have a slight preference for writing it in one line. But leaving it as is should also be OK with me.

**Review by yvonnefroehlich — APPROVED** [review]
I thinks the autocompletion for the `resolution` parameter is an improvement, as the available resolutions differ between the datasets.

**Pull request merged**